week3/homework/src/index.js

```diff
@@ -1,3 +1,150 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const uuidv4 = require('uuid/v4');
+const port = 3041;
+
+app.use(express.json());
+
+function readTodoFiles() {
+  return new Promise(function(resolve, reject) {
+    fs.readFile('todo.json', 'utf8', (err, data) => {
+      if (err) reject(err);
+      else resolve(JSON.parse(data));
+    });
+  });
+}
+
+function saveTodos(savePath) {
+  return new Promise(function(resolve, reject) {
+    fs.writeFile('todo.json', JSON.stringify(savePath), 'utf8', err => {
+      if (err) reject(err);
+      else resolve();
+    });
+  });
+}
+
+async function readTodos(req, res) {
+  const data = await readTodoFiles();
+  res.send(data);
+}
+
+async function deleteTodo(req, res) {
+  let todos = await readTodoFiles();
+  let todo = todos.find(todo => todo.id == req.params.id);
+  let indexOfTodo = todos.indexOf(todo);
+  todos.splice(indexOfTodo, 1);
+  todos.push(todo);
+  await saveTodos(todo);
+  res.status = 201;
+  res.send();
+}
+
+async function createTodo(req, res) {
+  const todo = validateAndParseTodo(req);
+  const todos = await readTodoFiles();
+  todo.id = uuidv4();
+  todo.done = false;
+  todos.push(todo);
+  await saveTodos(todos);
+  res.status = 201;
+  res.send();
+}
+
+async function updateTodo(req, res) {
+  try {
+    const updateTodo = validateAndParseTodo(req);
+    const todos = await readTodoFiles();
+    const originalTodo = todos.find(todo => todo.id === req.params.id);
+    originalTodo.description = updateTodo.description;
+    await saveTodos(todos);
+    res.send();
+  } catch (err) {
+    new Error(console.log(err.message));
+  }
+}
+
+function validateAndParseTodo(req) {
+  const { todo } = req.body;
+  if (todo == null) throw Error('todo not set');
+
+  if (todo.description != null) todo.description = todo.description.trim();
+
+  if (todo.description == null || todo.description.length === 0)
+    throw new Error('description not set');
+  return todo;
+}
+
+async function markAsDone(req, res) {
+  try {
+    const todos = await readTodoFiles();
+    const targetedTodo = todos.find(todo => todo.id === req.params.id);
+    targetedTodo.done = true;
+    await saveTodos(todos);
+    res.end('this task is done');
+  } catch (err) {
+    console.log(err.message);
+  }
+}
+
+async function markAsNotDone(req, res) {
+  const todos = await readTodoFiles();
+  const todo = todos.find(todo => todo.id === req.params.id);
+  todo.done = false;
+  await saveTodos(todos);
+  res.end('this task is not done');
+}
+
+async function readSpecificTodo(req, res) {
+  const todos = await readTodoFiles();
+  const task = todos.find(todo => todo.id === req.params.id);
+  res.send(task);
+}
+
+async function deleteAllTodos(req, res) {
+  let allData = await readTodoFiles();
+  allData = [];
+  await saveTodos(allData);
+  res.status = 201;
+  res.send();
```

Comment:
That being said you are not retuning any data on success, I would go for returning some kind of object just as a note to the client saying `Hey your request is succeed`, well it's happening in the status.

but just as a convention you could response with at least `{ ok: 'true' }` which makes it kinda success and same response type.